utils/checkdb.py

```diff
@@ -64,35 +91,47 @@ def check_tile_overlap(db, db_root, strict=False, verbose=False):
     Throw an exception if two tiles share an address
     '''
     mall = dict()
+    tiles_type_done = set()
+    db_files = dict()
 
     tiles_checked = 0
 
-    def subtiles(tile_names):
-        for tile_name in tile_names:
-            yield tile_name, db.tilegrid[tile_name]
-
     for tile_name, tilej in db.tilegrid.items():
-        # for tile_name, tilej in subtiles(["CLBLL_L_X14Y112", "INT_L_X14Y112"]):
+        tile_type = tilej["type"]
+
+        if tile_type not in tiles_type_done:
+            db_files = parsedb_file(db_root, db_files, tilej)
+            tiles_type_done.add(tile_type)
+
+        if tile_type not in mall:
+            verbose and print("Adding tile type: %s" % tile_type)
+            mall[tile_type] = {}
```

Comment:
Why not just use prjxray.db.Database and prjxray.grid.Grid?  Why rewrite the parsing code?

Reply:
Done, now the script uses the libraries.